decentraland/unity-renderer
# Chore: remove places and events prewarm

## What does this PR change?
Closes #3800 

### Problem: 
`FirstLoading` function of `Places/Events/HighlightSubSectionComponentController`s was called when `View.Ready` event was fired and it was then requesting data from API via `RequestAllEvents/Places/Highlgihts` method. With high chance, this view was ready even before Scene was fully loaded. 

### Solution:
I moved call for `FirstLoading` function from `View.Ready` event to `view.OnPlacesSubSectionEnable` event, so it is called when section is actually opened (without pre-warming)

### Additional changes:
1. Extracting repeated reloading logic into `PlaceAndEventsCardsReloader` class
2. Split Places and Events Utils classes to one common Factory and 2 separate Configurators classes
3. Some other refactoring of Places and Events controllers with removing of duplicated code by adding generic methods, extracting common interfaces, and applying strategy pattern (in functional way)

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=chore/remove-places-and-events-prewarm
2. Open browser profiling (F12) and go to Network section
3. Observe .jpg/.png files for events and places being loaded only when needed: 
    - in Genesis plaza, when scene is loaded, 
    - in other places, when ExploreV2 Menu for PlacesAndEvents section open.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
